### PR TITLE
io/lab: Explain UNIX sockets

### LINKED ISSUE #137: io/lab: Explain that UNIX sockets only work in POSIX systems
It's a pretty obvious thing, but it's not mentioned in the I/O lab: https://open-education-hub.github.io/operating-systems/Lab/IO/beyond-network-sockets#unix-sockets. It should be.

### PR DESCRIPTION
**Pull request by Alex-deVis**
Stating the obvious won't hurt.
Closes #137.

### CODE REVIEW

File: content/chapters/io/lab/content/beyond-network-sockets.md [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 ## UNIX Sockets
 
-UNIX sockets allow us to create them using the `socket()` syscall and `bind()` them **TO A FILE** instead of an IP and port.
+UNIX sockets are created using the `socket()` syscall and are bound **TO A FILE** instead of an IP and port using `bind()`.
 You may already see a similarity with [named pipes](./pipes.md#named-pipes---mkfifo).
 Just like them, UNIX sockets don't work by writing data to the file (that would be slow), but instead the kernel retains the data they send internally so that `send()` and `recv()` can read it from the kernel's storage.
 You can use `read()` and `write()` to read/write data from/to both network and UNIX sockets as well, by the way.
 The differences between using `send()`/`recv()` or `write()`/`read()` are rather subtle and are described in [this Stack Overflow thread](https://stackoverflow.com/questions/1790750/what-is-the-difference-between-read-and-recv-and-between-send-and-write).
 
+UNIX sockets are a feature of POSIX-compliant operating systems (e.g. Linux, macOS) and are not available on non-POSIX operating systems such as Microsoft Windows.
```
Comment (teodutu) [inline content/chapters/io/lab/content/beyond-network-sockets.md:21]:
```suggestion
UNIX sockets are a feature of POSIX-compliant operating systems (e.g. Linux, macOS) and are not available on non-POSIX operating systems, such as Microsoft Windows.
```

File: content/chapters/io/lab/content/beyond-network-sockets.md [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+However, there are third-party libraries providing similar features to UNIX sockets in non-POSIX systems, but they might not have the same performance and reliability.
```
Comment (teodutu) [inline content/chapters/io/lab/content/beyond-network-sockets.md:22]: It may be useful to provide an example.

**Review by teodutu — CHANGES_REQUESTED** [review]
In addition to my inline suggestions, you should squash your commits and add a description to satisfy checkpatch.

**Comment by Alex-deVis** [comment]
> In addition to my inline suggestions, you should squash your commits and add a description to satisfy checkpatch.

The checkpatch is satisfied.

**Comment by gabrielmocanu** [comment]
image
Checkpatch right now

**Review by teodutu — APPROVED** [review]

**Pull request merged**